### ISSUE TITLE
optimal_scale returns nan on degenerate input; the nan then silently corrupts fit_thickness and fit_orientation

### ISSUE THREAD

**Issue by imaitland** [Differentiable-Electron-Crystallography/diffBloch#43]
`optimal_scale` returns `nan` on degenerate input instead of raising, and the `nan` then silently corrupts the preprocess fits that consume it.

## The defect

`src/diffBloch/core/losses.py`, in `optimal_scale`:

```python
ratio = observed.sum() / calculated.sum()
```

Two degenerate inputs both produce `nan` with no error and no warning:

```python
import torch
from diffBloch.core.losses import optimal_scale, w_rbragg

empty = torch.zeros(0, dtype=torch.float64)
optimal_scale(empty, empty, empty, metric=w_rbragg)
# -> (nan, nan)          0/0 in the ratio

n = 5
observed = torch.rand(n, dtype=torch.float64) + 0.5
sigmas = torch.full((n,), 0.1, dtype=torch.float64)
optimal_scale(torch.zeros(n, dtype=torch.float64), observed, sigmas, metric=w_rbragg)
# -> (inf, nan)          calculated sums to zero
```

A healthy call returns `(0.5, 0.0)`, so the failure is silent rather than obviously wrong.

## The same degeneracy is already guarded elsewhere — but not here

`engine/losses.py::scaled_w_rbragg_loss`, the refinement objective, defends against exactly this:

```python
denom = calc.sum(dim=-1, keepdim=True).clamp_min(eps)   # guard tiny calc
```

and its docstring explicitly contrasts itself with `optimal_scale`. So the gradient path is protected
and its `no_grad` sibling is not, for the same failure, in the same codebase. That inconsistency is
the core of this issue.

## How the `nan` corrupts its consumers

`nan` does not stay local — it defeats the reductions the preprocess fits are built on:

```python
v = torch.tensor([0.5, float('nan'), 0.3, 0.4])
int(torch.argmin(v))    # 1  -- the nan index, not 0.3
float(torch.min(v))     # nan
float('nan') > 0.3      # False -- and so is every other comparison
```

Two concrete consequences:

**`fit_thickness` bakes a thickness chosen by a `nan`.** It grid-searches
`score_orientation_per_thickness` and takes `argmin`. One degenerate thickness candidate makes
`argmin` return *that* index, so the rotation is settled at whatever thickness produced the
degeneracy — silently, and the value is then carried by the `Plan` into everything downstream.

**`fit_orientation` stops being able to accept an improvement.** `score_orientation` is
`score_orientation_per_thickness(...).min()`, and `torch.min` propagates `nan`. A rotation scoring
`nan` compares `False` against every trial, so the hexagonal search's monotone-descent logic never
registers a better orientation and the rotation silently keeps its seed.

In both cases the output looks ordinary: a settled `Plan` with a thickness and an orientation, no
error, no warning, and nothing identifying which rotation went degenerate.

## Suggested fix

Guard at the source, in `optimal_scale`:

- raise on an empty reduction axis (an empty aligned set is a caller error, not a score of `nan`)
- raise, or apply the same `clamp_min` treatment `scaled_w_rbragg_loss` already uses, when
  `calculated.sum()` is zero

Whichever is chosen, the two paths should agree — right now the differentiable loss defends and the
`no_grad` scorer does not.

Worth considering separately whether the `argmin` / `min` reductions in the fits should reject a
non-finite score outright, so that a `nan` arriving from any future source fails loudly rather than
being silently selected.
